dotnet/src/Microsoft.AutoGen/Agents/InMemory/InMemoryAgentWorkerRuntime.cs

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// InMemoryAgentWorkerRuntime.cs
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using Microsoft.AutoGen.Abstractions;
+using Microsoft.AutoGen.Runtime;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AutoGen.Agents;
+
+public class InMemoryAgentWorkerRuntime : IAgentWorkerRuntime, IAgentWorkerRegistryGrain, IWorkerGateway, IAgentContext
+{
+    private readonly ILogger<InMemoryAgentWorkerRuntime> _logger;
+    private readonly InMemoryQueue<CloudEvent> _eventsQueue = new();
+    private readonly InMemoryQueue<Message> _messageQueue = new();
+    private readonly ConcurrentDictionary<string, AgentState> _agentStates = new();
+    private readonly ConcurrentDictionary<string, Type> _agentTypes = new();
+    private readonly Dictionary<string, List<IWorkerGateway>> _supportedAgentTypes = [];
+    private readonly Dictionary<IWorkerGateway, WorkerState> _workerStates = [];
+    private readonly ConcurrentDictionary<string, IWorkerGateway> _workers = new();
+    private readonly ConcurrentDictionary<string, (IAgentBase Agent, string RequestId)> _pendingRequests = new();
+    private readonly ConcurrentDictionary<AgentId, IWorkerGateway> _agentPlacements = new();
+
+    AgentId IAgentContext.AgentId => throw new NotImplementedException();
+
+    IAgentBase? IAgentContext.AgentInstance { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+    DistributedContextPropagator IAgentContext.DistributedContextPropagator => throw new NotImplementedException();
+
+    ILogger IAgentContext.Logger => _logger;
```

Comment:
Logger shouldn't be part of abstraction?

Reply:
removed